[benchmark] WordCount: Initialize global variables in setUpFunction.

lorentey opened the pull request:
WordCount shows significant variance between iterations, which is apparently caused by the initialization of someAlphanumerics and asciiWords/utf16Words leaking into the measured part of the benchmark.

Making sure these variables are initialized before we start measuring elapsed time stabilizes the results.

lorentey committed:
[benchmark] WordCount: Initialize global variables in setUpFunction.

WordCount shows significant variance between iterations, which is apparently caused by the initialization of someAlphanumerics and asciiWords/utf16Words leaking into the measured part of the benchmark.

Making sure these variables are initialized before we start measuring elapsed time stabilizes the results.

lorentey commented:
@swift-ci smoke test

lorentey commented:
@swift-ci smoke benchmark

lorentey committed:
[benchmark] WordCount: Also blackHole WordSplit tests, for good measure

lorentey commented:
@swift-ci smoke test

lorentey commented:
@swift-ci smoke test

lorentey commented:
@swift-ci smoke benchmark

lorentey commented:
Oh great, the bot insists on testing 872efe1 again.

@swift-ci smoke benchmark

milseman reviewed (approved):
I would just make a function `buildWorkload()` that blackHoles all of these, rather than try to decide which apply to which sub-benchmark.

(e.g. CSVParsing.swift follows this approach)

Otherwise LGTM

lorentey committed:
[benchmark] WordCount: Use a unified buildWorkload() func

setUpFunction code should be located together with the corresponding runFunction code, so that we can verify them at a glance. However, we can't really do that yet, and doing the setUp for all tests in bulk makes it less likely that we make a mistake in matching up tests with their corresponding setups. :/

lorentey commented:
@swift-ci smoke-test and merge

milseman commented:
@swift-ci please smoke test and merge

milseman commented:
@swift-ci please smoke test

milseman commented:
@swift-ci `defer { merge() }`
